mobile/components/views/Attractions/AddNew/OpeningHour.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect } from 'react';
+import { View, Text, Modal, TextInput, Button, TouchableOpacity } from 'react-native';
+
+const OpeningHoursComponent = ({ label, value, onChangeText }) => (
+  <>
+    <Text>{label}</Text>
+    <TextInput
+      style={{ height: 40, borderColor: 'gray', borderWidth: 1, marginBottom: 10 }}
+      keyboardType="numeric"
+      value={value.toString()}
+      onChangeText={onChangeText}
+    />
+  </>
+);
+
+const OpeningHoursModal = ({ visible, onClose, onSave, initialOpeningHours }) => {
+  const [selectedDayIndex, setSelectedDayIndex] = useState(0);
+  const getDefaultOpeningHours = () => {
+    return [
+      { day: 'Mon', openingHour: '09:00', closingHour: '17:00' },
+      { day: 'Tue', openingHour: '09:00', closingHour: '17:00' },
+      { day: 'Wed', openingHour: '09:00', closingHour: '17:00' },
+      { day: 'Thu', openingHour: '09:00', closingHour: '17:00' },
+      { day: 'Fri', openingHour: '09:00', closingHour: '17:00' },
+      { day: 'Sat', openingHour: '09:00', closingHour: '17:00' },
+      { day: 'Sun', openingHour: '09:00', closingHour: '17:00' },
+    ];
+  };
+  const [openingHours, setOpeningHours] = useState(initialOpeningHours || getDefaultOpeningHours());
```

Comment:
It can possibly crash because its gonna be loaded asynchronously from parent component, I would move it to useEffect and there check if `initialOpeningHours ` exists and then change `openingHours` state